tests/unit/backend/layers/api/test_curation_api.py

```diff
@@ -83,13 +83,21 @@ def test__delete_private_collection_by_collection_version_id(self):
                 private_collection_version_id = self.generate_unpublished_collection().version_id
                 self._test(private_collection_version_id, auth, expected_response)
 
-    def test__delete_tombstone_collection(self):
+    @patch("backend.layers.thirdparty.s3_provider.S3Provider.delete_files")  # required for tombstoning
+    def test__delete_tombstone_collection(self, mock_delete: Mock):
         tests = [("not_owner", 403), ("noauth", 401), ("owner", 403), ("super", 403)]
         for auth, expected_response in tests:
             with self.subTest(auth):
-                collection_id = self.generate_published_collection().collection_id
-                self.business_logic.tombstone_collection(collection_id)
-                self._test(collection_id, auth, expected_response)
+                collection = self.generate_published_collection()
+                self.business_logic.tombstone_collection(collection.collection_id)
+                if self.run_as_integration:
```

Comment:
no need to complicate the db mock code to handle this, so we'll just run the arg check for integration tests only